CMakeLists.txt

```diff
@@ -56,297 +63,80 @@ zephyr_include_directories(
-set_ifndef(DIALECT_STD_CPP11 "c++11")
-set_ifndef(DIALECT_STD_CPP14 "c++14")
-set_ifndef(DIALECT_STD_CPP17 "c++17")
-set_ifndef(DIALECT_STD_CPP2A "c++2a")
-
-if(CONFIG_STD_CPP98)
-  set(STD_CPP_DIALECT ${DIALECT_STD_CPP98})
-elseif(CONFIG_STD_CPP11)
-  set(STD_CPP_DIALECT ${DIALECT_STD_CPP11}) # Default
-elseif(CONFIG_STD_CPP14)
-  set(STD_CPP_DIALECT ${DIALECT_STD_CPP14})
-elseif(CONFIG_STD_CPP17)
-  set(STD_CPP_DIALECT ${DIALECT_STD_CPP17})
-elseif(CONFIG_STD_CPP2A)
-  set(STD_CPP_DIALECT ${DIALECT_STD_CPP2A})
-else()
-  assert(0 "Unreachable code. Expected C++ standard to have been chosen. See misc/Kconfig.")
-endif()
-
-zephyr_compile_options(
-  ${OPTIMIZATION_FLAG} # Usually -Os
-  -g # TODO: build configuration enough?
-  -Wall
-  -Wformat
-  -Wformat-security
-  -Wno-format-zero-length
-  -imacros ${AUTOCONF_H}
-  -ffreestanding
-  -Wno-main
-  ${NOSTDINC_F}
-  ${TOOLCHAIN_C_FLAGS}
-)
-
-zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-std=${STD_CPP_DIALECT}>
-  $<$<COMPILE_LANGUAGE:CXX>:-fcheck-new>
-  $<$<COMPILE_LANGUAGE:CXX>:-ffunction-sections>
-  $<$<COMPILE_LANGUAGE:CXX>:-fdata-sections>
-
-  $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
-  $<$<COMPILE_LANGUAGE:ASM>:-D_ASMLANGUAGE>
-)
-
-if(NOT CONFIG_RTTI)
-zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
-)
-endif()
-
-if(NOT CONFIG_EXCEPTIONS)
-zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-fno-exceptions>
-)
-endif()
-
-if(NOT CONFIG_NATIVE_APPLICATION)
-zephyr_ld_options(
-  -nostdlib
-  -static
-  -no-pie
-)
-endif()
-
-if(CONFIG_LIB_CPLUSPLUS)
-zephyr_ld_options(
-  -lstdc++
-)
-endif()
-
-# ==========================================================================
-#
-# cmake -DW=... settings
-#
-# W=1 - warnings that may be relevant and does not occur too often
-# W=2 - warnings that occur quite often but may still be relevant
-# W=3 - the more obscure warnings, can most likely be ignored
-# ==========================================================================
-if(W MATCHES "1")
-  zephyr_compile_options(
-    -Wextra
-    -Wunused
-    -Wno-unused-parameter
-    -Wmissing-declarations
-    -Wmissing-format-attribute
-    -Wold-style-definition
-    )
-  zephyr_cc_option(
-    -Wmissing-prototypes
-    -Wmissing-include-dirs
-    -Wunused-but-set-variable
-    -Wno-missing-field-initializers
-    )
-endif()
-
-if(W MATCHES "2")
-  zephyr_compile_options(
-    -Waggregate-return
-    -Wcast-align
-    -Wdisabled-optimization
-    -Wnested-externs
-    -Wshadow
-    )
-  zephyr_cc_option(
-    -Wlogical-op
-    -Wmissing-field-initializers
-    )
-endif()
-
-if(W MATCHES "3")
-  zephyr_compile_options(
-    -Wbad-function-cast
-    -Wcast-qual
-    -Wconversion
-    -Wpacked
-    -Wpadded
-    -Wpointer-arith
-    -Wredundant-decls
-    -Wswitch-default
-    )
-  zephyr_cc_option(
-    -Wpacked-bitfield-compat
-    -Wvla
-    )
-endif()
+# Discover if the default C standard has been overridden
+# So far, only POSIX-based boards does this
```

Comment:
> So far, only POSIX-based boards does this

This comment is not likely to be updated, and should therefore be dropped, which is unfortunate, because it is useful information.